### DISCUSSION TITLE
Looping over pandas dataframe with track()

### DISCUSSION THREAD

**Discussion by astrochun** [Textualize/rich#2202] [Q&A] [answered]
I'm trying to figure out what the best way is to run a script that loops over a `pandas` DataFrame with `rich.progress.track`.

Since `.iterrows()` and `itertuples()`, provide generators instead of iterators, I came up with:

```python
for _t in track(list(df.itertuples()), description='processing...'):
     print(_t[0])  # print index
     ...
```

Any other way?

**Comment by willmcgugan** [comment, marked as answer]
You could use `itertuples` directly, but you would need to supply a `total` argument so Rich knows how many items it contains.

**Reply by astrochun** [reply]
I realized what I did wrong when I tried to specify the `total` argument:
```
for _t in track(df.itertuples, description='processing...', total=len(df)):
     ...
```

I failed to call the `itertuple` method. Just thought I mentioned it here since others might encounter the same issue. There's very little documentation beyond basic (e.g. range or list) examples. If I get a chance, I'll create a PR.

**Reply by ju674** [reply]
for _t in track(df.itertuples, description='processing...', total=len(df)):
     ...
